manual_flagging/flagging.py

```diff
@@ -0,0 +1,517 @@
+from glue.flagging_script_glue import flagging as flg_model
+from glue import sales_val_flagging as flg
+import awswrangler as wr
+import copy
+import os
+import datetime
+import numpy as np
+import pandas as pd
+import pytz
+import subprocess as sp
+import yaml
+from pyathena import connect
+from pyathena.pandas.util import as_pandas
+
+# Set working dir to manual_update, standardize yaml and src locations
+root = sp.getoutput("git rev-parse --show-toplevel")
+os.chdir(os.path.join(root, "manual_flagging"))
+
+# Use yaml as inputs
+with open(os.path.join("yaml", "inputs.yaml"), "r") as stream:
+    inputs = yaml.safe_load(stream)
+
+# Validate the input specification
+try:
+    # Check housing_market_type
+    assert "housing_market_type" in inputs, "Missing key: 'housing_market_type'"
+    assert set(inputs["housing_market_type"]).issubset(
+        {"res_single_family", "res_multi_family", "condos"}
+    ), "housing_market_type can only contain 'res_single_family', 'res_multi_family', 'condos'"
+    assert len(inputs["housing_market_type"]) == len(
+        set(inputs["housing_market_type"])
+    ), "Duplicate values in 'housing_market_type'"
+
+    # Check run_tri
+    assert "run_tri" in inputs, "Missing key: 'run_tri'"
+    assert set(inputs["run_tri"]).issubset(
+        {1, 2, 3}
+    ), "run_tri can only contain 1, 2, 3"
+    assert len(inputs["run_tri"]) == len(
+        set(inputs["run_tri"])
+    ), "Duplicate values in 'run_tri'"
+
+except AssertionError as e:
+    print(f"Error: {e}")
```

Comment:
This chunk is used to validate proper values are specified in the run filters in the `housing_market_type` and `run_tri` lists of the `inputs.yaml` file. It checks for valid values and duplicates.